src/structural/organparameter.cpp

```diff
@@ -329,8 +330,9 @@ void OrganRandomParameter::readSuccessor(tinyxml2::XMLElement* p, bool verbose)
 		lookfor = std::vector<std::string>{"probability","percentage"};//probability (or percentage for backward compatibility)
 		defaultVald = successorNo.at(ruleId)/successorST.at(ruleId).size();//default = number of laterals / option for laterals
 		defaultSize = (successorST.at(ruleId).size() - successorP.at(ruleId).size());
-
+#ifndef _MSC_VER // MSVC will instead throw a division by zero error
 		if(!std::isfinite(defaultVal))
+#endif
```

Comment:
won t this throw an issue if _MSC_VER is defined? there is nothing to replace the "if" condition. also that s actually an error I put in: it should be if(!std::isfinite(defaultVald)) and not if(!std::isfinite(defaultVal)).
if _MSC_VER is defined you can put "if(successorST.at(ruleId).size() == 0)". I don t see other obvious reasons why defaultVald would be infinite.

Reply:
Ah okay that makes sense I was not even looking at the difference there because skim-reading the code when MSVC threw the error I didn't see that there was a floating point value that is similarly named. Using the correct variable there would also remove the need for that distinction. Weird that neither clang nor gcc threw errors there though.